native/src/main/java/io/ballerina/stdlib/mqtt/client/ClientActions.java

```diff
@@ -43,7 +68,37 @@ public static Object externInit(BObject clientObject, BString serverUri, BString
             MqttClient publisher = new MqttClient(serverUri.getValue(), clientId.getValue(), new MemoryPersistence());
             MqttConnectionOptions options = MqttUtils.getMqttConnectOptions(clientConfiguration);
             publisher.connect(options);
-            clientObject.addNativeData(MqttConstants.CLIENT_OBJECT, publisher);
+            clientObject.addNativeData(MqttConstants.MQTT_CLIENT, publisher);
+            LinkedBlockingQueue blockingQueue = new LinkedBlockingQueue<>();
+            LinkedBlockingQueue deliveryTokenQueue = new LinkedBlockingQueue<>();
+            ExecutorService executor = Executors.newCachedThreadPool();
+            clientObject.addNativeData(RESPONSE_QUEUE, blockingQueue);
+            clientObject.addNativeData(DELIVERY_TOKEN_QUEUE, deliveryTokenQueue);
+            clientObject.addNativeData(RESPONSE_EXECUTOR_SERVICE, executor);
+            publisher.setCallback(new MqttCallback() {
```

Comment:
Shall we write a separate class for this `MqttCallback` implementation and use it here ?

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-mqtt/pull/9/commits/041d5b9016d02da09f39901fe4d4970544edfb25